### PR TITLE
Feat: support parameter in health check of definition

### LINKED ISSUE #4863: [Feature] support parameter in health evaluation cue template
**Is your feature request related to a problem? Please describe.**

I need to define multiple traits of the SAME TYPE to the same component. Thus I need to get the output of that specific trait by its name. Something the folllowing `healthPolicy` but it is not working… What can I do?

**Describe the solution you'd like**

```
apiVersion: core.oam.dev/v1beta1
kind: TraitDefinition
metadata:
  name: mytrait
spec:
  schematic:
    cue:
      template: |
        outputs: "mytrait-\(parameter.name)": {
          // ...
        }
        parameter: {
          name: string
        }
  status:
    healthPolicy: 'isHealth: context.outputs."mytrait-\(parameter.name)".status.state == "Available"'
```

**Describe alternatives you've considered**

**Additional context**

From slack https://cloud-native.slack.com/archives/C01BLQ3HTJA/p1665675028027679 by Manuel Coppotelli

### PR DESCRIPTION
**Pull request by wonderflow**
Signed-off-by: Jianbo Sun <jianbo.sjb@alibaba-inc.com>

### Description of your changes

Fixes #4863

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/appfile/appfile.go [hunk 1]
```diff
@@ -114,7 +114,7 @@ func (wl *Workload) EvalHealth(ctx process.Context, client client.Client, access
 	if wl.SkipApplyWorkload {
 		return true, nil
 	}
-	return wl.engine.HealthCheck(ctx, client, accessor, wl.FullTemplate.Health)
```
Comment (barnettZQG) [inline pkg/appfile/appfile.go:117]: Please add some tests to cover this new feature.

File: pkg/cue/definition/template.go [hunk 2]
```diff
@@ -205,25 +205,44 @@ func (wd *workloadDef) getTemplateContext(ctx process.Context, cli client.Reader
 	return root, nil
 }
 
+func formRuntimeContext(templateContext map[string]interface{}, parameter interface{}) (string, error) {
```
Comment (FogDong) [inline pkg/cue/definition/template.go:208]:
```suggestion
func fromRuntimeContext(templateContext map[string]interface{}, parameter interface{}) (string, error) {
```

File: pkg/appfile/appfile.go (hunk 1 above)
Comment (wonderflow) [reply pkg/appfile/appfile.go:117]: added

**Review by barnettZQG — APPROVED** [review]
LGTM

**Review by Somefive — APPROVED** [review]

**Pull request merged**